lib/src/mocks/http.dart

```diff
@@ -94,15 +111,21 @@ class MockHttp {
   }
 
   void expect(String method, Uri uri,
-      {Object failWith, WResponse respondWith}) {
+      {Object failWith, Map<String, String> headers, WResponse respondWith}) {
     if (failWith != null && respondWith != null) {
       throw new ArgumentError('Use failWith OR respondWith, but not both.');
     }
     if (failWith == null && respondWith == null) {
       respondWith = new MockWResponse.ok();
     }
-    _expectations.add(new _RequestExpectation(method, uri,
-        failWith: failWith, respondWith: respondWith));
+    _expectations.add(new _RequestExpectation(
+        headers == null
+            ? new CaseInsensitiveMap()
```

Comment:
Converting a null value for `headers` to an empty map ruins the check at line 46 above. The idea is that if `headers` is null, then the tester doesn't care about them and will match requests based only on URI and method. By converting this to an empty map, the tester won't match requests that have headers..

```dart
MockTransports.http.expect('GET', Uri.parse('/path'));
WHttp.get(Uri.parse('/path'), headers: {'x-custom': 'value'});
```
I would expect this request to resolve because my `.expect()` call didn't specify headers.

```dart
MockTransports.http.expect('GET', Uri.parse('/path'), headers: {});
WHttp.get(Uri.parse('/path'), headers: {'x-custom': 'value'});
```
I would expect this request not to resolve since I explicitly stated an expectation of no headers.